frontend/src/metabase/query_builder/components/view/QuestionDataSource.unit.spec.js

```diff
@@ -282,7 +230,16 @@ class ErrorBoundary extends Component {
 }
 const SOURCE_CARD = createMockCard({ id: SOURCE_QUESTION_ID });
 
-function setup({ question, subHead = false, isObjectDetail = false } = {}) {
+function setup({
+  card,
+  subHead = false,
+  isObjectDetail = false,
+  hasPermissions = true,
+  isMultiSchemaDB = false,
+} = {}) {
+  const metadata = hasPermissions ? getMetadata() : {};
```

Comment:
1. Let's make sure `metadata` is a valid instance of `Metadata`

```suggestion
  const metadata = hasPermissions ? getMetadata() : createMockMetadata({});
```

2. I'd make `metadata` a parameter of `setup` (instead  `hasPermissions`) - this gives even greater control. But maybe YAGNI.